packages/components/src/components/dialog/dialog.jsx

```diff
@@ -17,20 +17,47 @@ const Dialog = ({
     is_visible,
     onCancel,
     onConfirm,
+    onEscapeButtonCancel,
     ...other_props
 }) => {
+
+    const {
+        cancel_button_text,
+        className,
+        children,
+        confirm_button_text,
+        is_loading,
+        is_mobile_full_width = true,
+        is_content_centered,
+        portal_element_id,
+        title,
+        has_close_icon,
+    } = other_props;
+
     const wrapper_ref = React.useRef();
+
     React.useEffect(() => {
         if (is_visible && !!disableApp) {
             disableApp();
         }
     }, [is_visible, disableApp]);
 
+    React.useEffect(() => {
+        const close = (e) => {
+          if(e.key === 'Escape'){
+            handleCancel();
+          }
+        }
+        window.addEventListener('keydown', close);
+      return () => window.removeEventListener('keydown', close);
+    },[])
```

Comment:
@george-usynin-deriv Could you fill the dependency array or add the diasable eslint warning comment.
Also, we expect to have a semicolon at the end of the line.